scripts/components/Dialog/PasswordContent.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { H5PContext } from "../../context/H5PContext";
+import './PasswordContent.scss';
+
+export default class PasswordContent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      unlocked: false,
+      hasClicked: false,
+      shakeClass : "",
+      showPasswordInput : true
+    }
+  }
+
+  handleOnClick = event => {
+    event.preventDefault();
+    this.setState({
+      hasClicked: true
+    })
+    if(this.props.currentInteraction.unlocked) {
+      this.props.showInteraction(this.props.currentInteractionIndex)
+    } else {
+      this.setState({
+        unlocked: this.props.handlePassword(this.input.value)
+      })
+      if(!this.props.currentInteraction.unlocked){
+        this.shakeIcon();
+
+      }
+    }
+  };
+  toggleShowCode = () => {
+
+    this.setState({
+      showPasswordInput : !this.state.showPasswordInput
+    })
+
+  }
+
+  shakeIcon = () => {
+    this.setState({
+      shakeClass : "h5p-password-icon--shake"
+    })
+    setTimeout((() => {
+      this.setState({
+        shakeClass : ""
+      })
+    }), 500)
+  };
+  render() {
+//TODO: replace with l10n translation
```

Comment:
What's being replaced with l10n translation?

Reply:
Just an old reminder to replace hardcoded strings with l10n vars